pkg/remote/aws/apigatewayv2_route_enumerator.go

```diff
@@ -0,0 +1,50 @@
+package aws
+
+import (
+	"github.com/cloudskiff/driftctl/pkg/remote/aws/repository"
+	remoteerror "github.com/cloudskiff/driftctl/pkg/remote/error"
+	"github.com/cloudskiff/driftctl/pkg/resource"
+	"github.com/cloudskiff/driftctl/pkg/resource/aws"
+)
+
+type ApiGatewayV2RouteEnumerator struct {
+	repository repository.ApiGatewayV2Repository
+	factory    resource.ResourceFactory
+}
+
+func NewApiGatewayV2RouteEnumerator(repo repository.ApiGatewayV2Repository, factory resource.ResourceFactory) *ApiGatewayV2RouteEnumerator {
+	return &ApiGatewayV2RouteEnumerator{
+		repository: repo,
+		factory:    factory,
+	}
+}
+
+func (e *ApiGatewayV2RouteEnumerator) SupportedType() resource.ResourceType {
+	return aws.AwsApiGatewayV2RouteResourceType
+}
+
+func (e *ApiGatewayV2RouteEnumerator) Enumerate() ([]*resource.Resource, error) {
+	apis, err := e.repository.ListAllApis()
+	if err != nil {
+		return nil, remoteerror.NewResourceListingError(err, string(e.SupportedType()))
```

Comment:
Here we should return an error that means that it is the listing of the Api Gateway V2 Apis that failed and not the listing of the Routes.

`return nil, remoteerror.NewResourceListingErrorWithType(err, string(e.SupportedType()), aws.AwsApiGatewayV2ApiResourceType)`